### PR TITLE
Do noy allow empty urls to be submitted in "add to your owncloud"

### LINKED ISSUE #19382: Federated sharing: Strange error when not putting in address
When putting in no address and pressing enter (or the confirm button), you get this strange error:
federated-sharing-no-url-put

When no URL is put in, nothing should happen (the button should be disabled). @schiesbn

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #19382

CC: @PVince81 @jancborchardt @MorrisJobke

### CODE REVIEW

**Comment by karlitschek** [comment]
makes sense :+1:

**Comment by MorrisJobke** [comment]
Pasting with middle-mouse click (on linux select a text and then do a middle mouse click to paste the selected text) doesn't work anymore.

**Comment by MorrisJobke** [comment]
And we need a style for disabled icons. Simply the highlight should not be present.

File: apps/files_sharing/js/public.js [hunk 1]
```diff
@@ -228,6 +228,14 @@ OCA.Sharing.PublicApp = {
 			OCA.Sharing.PublicApp._saveToOwnCloud(remote, token, owner, name, isProtected);
 		});
 
+		$('#remote_address').on("keyup paste", function() {
```
Comment (MorrisJobke) [inline apps/files_sharing/js/public.js]: Adding `change` should do the trick

Comment (rullzer) [inline apps/files_sharing/js/public.js]: Done. seems to work here. 
@MorrisJobke could you double check?

**Comment by MorrisJobke** [comment]
I adjusted the style and remove the change event, because it didn't solve the problem with inserting via middle mouse click. I also think this is not that much of a problem.
(somehow the insert of the text takes place after the event is fired and then the value is still empty)

It's fine now with me :+1:

**Comment by icewind1991** [comment]
:+1: tested, code looks good

**Pull request merged**